tools/src/h5dump/h5dump.c

```diff
@@ -1051,11 +1053,41 @@ parse_command_line(int argc, const char *const *argv)
                 h5tools_nCols = 0;
                 break;
 
-            case 'm':
+            case 'm': {
+                char *format_specifier, *plus_flag;
```

Comment:
h5dump's "-m" option allows specifying the printf specifier for `float` and `double` data, as well as `long double` data if `sizeof(long double) == sizeof(double)`. Rather than having a separate flag for `float/double _Complex` data and a possible additional one for `long double _Complex` data, I just re-used this flag. A better solution is likely needed if XML output will continue to be supported, since the "-m" flag applying to `long double` (`_Complex`) data or not, depending on the platform, makes portable h5dump tests difficult.